### ISSUE TITLE
Introduce common tests between this gem and jsonapi-authorization?

### ISSUE THREAD

**Issue by valscion** [togglepro/pundit-resources#22]
Hi there,

I was looking at tests over at `jsonapi-authorization` side and I'm getting a feeling that they might be a bit out of hand. I have looked through your tests quite a lot, and checked out #21 for more than once (looking good by the way :+1:).

I'm impressed by the way you've done testing here, and I cannot shake the feeling of `jsonapi-authorization` and `pundit-resources` specs being quite similar in nature when come to the bottom of it. I was wondering whether I would be able to extract a subset of these specs as ones where both of our gems do it similarly.

Would you be interested in accepting a PR where I'd try to create some subset of tests that both of our gems could run? These common tests would most likely be run on request level, rather than on controller level. I'd try to keep those tests focused and not require library changes on your side.

I have a feeling that getting to a similar testing level would benefit both of our gems, and we'd be able to really see where our approaches diverge. It might even be that we'll both figure out what tradeoffs our gems do.

What do you think?

**Comment by barelyknown** [comment]
I'm not really opposed to it, but I'm also a bit wary of taking on any work related to `jsonapi-authorization` since I'm not familiar at all with the library (and probably won't be soon). So, if it's all the same to `pundit-resources`, fire away, but otherwise I think I'll pass.

**Comment by valscion** [comment]
Yeah, what I'd do would most likely be just some new specs that could be similar for both JA and pundit-resources. I'm not expecting any commitment from you, just wanted to hear whether you'd be opposed to the idea or not before I do anything about this :smile: Thanks!
